.github/actions/pr-comment/action.yml

```
@@ -60,8 +69,8 @@ runs:
         BODY: |
           ${{ inputs.comment }}
       run: |
-        if [[ ${{ inputs.edit-last }} == "true" ]]; then
-          echo "link=$(gh pr comment ${{ env.PR }} --edit-last --body '${{ env.BODY }}')" >> $GITHUB_OUTPUT
-        else
-          echo "link=$(gh pr comment ${{ env.PR }} --body '${{ env.BODY }}')" >> $GITHUB_OUTPUT
+        if [[ -n "${{ inputs.comment }}" ]]; then
+          echo "link=$(gh pr comment ${{ env.PR }} ${{ inputs.edit-last == 'true' && '--edit-last' || '' }} --body '${{ env.BODY }}' --repo ${{ github.repository }})" >> $GITHUB_OUTPUT
+        elif [[ -n "${{ inputs.comment-file }}" ]]; then
+          echo "link=$(gh pr comment ${{ env.PR }} ${{ inputs.edit-last == 'true' && '--edit-last' || '' }} --body-file "${{ inputs.comment-file }}" --repo ${{ github.repository }})" >> $GITHUB_OUTPUT
```

```suggestion
          echo "link=$(gh pr comment ${{ env.PR }} ${{ inputs.edit-last == 'true' && '--edit-last' || '' }} --body-file '${{ inputs.comment-file }}' --repo ${{ github.repository }})" >> $GITHUB_OUTPUT
```
Should it be single quotes here?

---

I think not - just in case we use an environment variable for a path to the file (for example, `$GITHUB_WORKSPACE/file.txt`

---

Ooh yup that makes sense. I think I wondered if having the `"` would match with the first quote at  start of `"link=..` so it was passing multiple arguments to echo rather than one string. But it would still just be one output line with echo so it'll maybe work fine?

---

Luckily, the quotations 'reset' inside a command (`$( ... )`) so we are allowed to do something like `echo "$(echo "")"` withough quoting issues!